metrics/otel_metrics.go

```diff
@@ -105,10 +107,19 @@ func (o *OTelMetrics) Start() error {
 		return err
 	}
 
+	// Fetch the hostname once and add it as an attribute to all metrics
+	var hostname string
+	if hn, err := os.Hostname(); err != nil {
+		hostname = "unknown: " + err.Error()
+	} else {
+		hostname = hn
+	}
+
 	res, err := resource.New(ctx,
 		resource.WithAttributes(resource.Default().Attributes()...),
 		resource.WithAttributes(attribute.KeyValue{Key: "service.name", Value: attribute.StringValue("refinery")}),
 		resource.WithAttributes(attribute.KeyValue{Key: "service.version", Value: attribute.StringValue(o.Version)}),
+		resource.WithAttributes(attribute.KeyValue{Key: "hostname", Value: attribute.StringValue(hostname)}),
```

Comment:
Can we use the semantic convention name `host.name`: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/resource/host.md#host

Reply:
we could add that as a duplicate, but there are TONS of boards and queries using "hostname". I'd be very reluctant to change it. I suppose we could do it differently for OTel metrics vs legacy metrics, but we'd definitely have to tell people we did it.

Reply:
I forgot the other metrics were already using hostname. Duplicating is best.